docs/terms/evaluation_parameter.md

```diff
@@ -1,15 +1,132 @@
+
+### Relationship to other objects
+
+Evaluation Parameters are used in Expectations when Validating data.  <TechnicalTag relative="../" tag="checkpoint" text="Checkpoints" /> use <TechnicalTag relative="../" tag="validation_action" text="Action" /> to store Evaluation Parameters in the <TechnicalTag relative="../" tag="evaluation_parameter_store" text="Evaluation Parameter Store" />.
+
+## Use cases
+
+<CreateHeader/>
+
+When creating expectations through the interactive method, Evaluation Parameters can be provided a temporary initial value.  This will allow you to test Expectations that are meant to rely on values from previous Validation runs before you have actually used them to Validate data.  Say you have created a Pandas dataframe called `my_df` that references a CSV that contains ten rows of data.  (To do so in the interactive process for creating Expectations, you would use the `great_expectations.read_csv()` method).  You want to create an expression that asserts that the row count for each Validation remains the same as the previous `upstream_row_count`, but since there is no previous `upstream_row_count` you need to provide a value that matches what the Expectation you are creating will find.
+
+To do so, you would define the parameter in question (`upstream_row_count`) by assigning it to the `$PARAMETER` value in a dictionary.  Then, you would provide the temporary value for that parameter by setting it as the value of the `$PARAMETER.<parameter_in_question>` key in the same dictionary.  Or, in this case, the `$PARAMETER.upstream_row_count`.
+
+For an example of this, see below:
+
+```python title="Python code"
+my_df.expect_table_row_count_to_equal(
+    value={"$PARAMETER": "upstream_row_count", "$PARAMETER.upstream_row_count": 10},
+    result_format={'result_format': 'BOOLEAN_ONLY'}
+)
+```
+
+This will return `{'success': True}`.
+
+An alternative method of defining the temporary value for an Evaluation Parameter is the `set_evaluation_parameter()` method, as shown below:
+
+```python title="Python code"
+my_df.set_evaluation_parameter("upstream_row_count", 10)
+```
+
+If the Evaluation Parameter's value is set in this way, you do not need to set it again (or define it alongside the use of the `$PARAMETER` key) for future Expectations.
+
+More typically, when validating Expectations, you can provide Evaluation Parameters that are only available at runtime.
+
+```python title="Python code"
+my_df.validate(
+    expectation_suite=my_dag_step_config, 
+    evaluation_parameters={"upstream_row_count": upstream_row_count}
+)
+```
+
+<ValidateHeader/>
+
+Evaluation Parameters that are configured as part of a Checkpoint's Expectations will be used without further interaction from you.  Additionally, Evaluation Parameters will be stored by having the `StoreEvaluationParametersAction` subclass of the `ValidationAction` class defined in a Checkpoint configuration's `action_list`.
```

Comment:
Checkpoints also have an object named evaluation_parameters -- could we add some detail about that? I haven't seen anyone use them that way, and would be curious to know more.

Reply:
I'm actually pretty unfamiliar with it myself -- this may be something I'll need to consult with @jcampbell on.

Reply:
Definitely worth mentioning. The evaluation parameters field in a checkpoint definition makes it possible to provide evaluation parameters at when the checkpoint runs, either via a dictionary or, for example, as an environment variable:

(UNTESTED OFF TOP OF HEAD EXAMPLE)

Suppose we have an expectation built like:
```
suite.add_expectation(
  ExpectationConfiguration(
    expectation_type="expect_table_row_count_to_equal",
    kwargs={
        "value": {"$PARAMETER": "upstream_row_count"}
    }
  )
)
```

```
name: my_checkpoint
...
evaluation_parameters:
  upstream_row_count: $MY_ENV_VAR
```

```
context.run_checkpoint('foo', evaluation_parameters={"upstream_row_count": 42})
```

Reply:
I've verified that these examples work and have added them into the document.